### PR TITLE
refactor: add nic_list_status to separate the computed values

### LINKED ISSUE #12: nutanix_virtual_machine, ip_endpoint_list not correctly working
from @BartJoris on https://github.com/nutanix/terraform-provider-nutanix/issues/205

May have been fixed by https://github.com/nutanix/terraform-provider-nutanix/pull/206, but need to validate

**Describe the bug**
When you specify ip_endpoint_list for nutanix_virtual_machine. It works fine, but if you rerun "terraform apply" it gives an error.

```bash
  nic_list = [{
    subnet_reference = {
      kind = "subnet"
      uuid = "${data.nutanix_subnet.external.metadata.uuid}"
    }

    ip_endpoint_list = {
      ip   = "${var.ip_start}${+count.index}"
      type = "ASSIGNED"
    }
  }]
```
**Expected behavior**
Rerunning "terraform apply" should not be updating the NIC list

**Logs**
```bash
data.nutanix_image.centos: Refreshing state...
data.nutanix_subnet.external: Refreshing state...
nutanix_virtual_machine.k8master: Refreshing state... (ID: c28a4244-43b9-4aeb-88fa-800621f79939)

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  ~ nutanix_virtual_machine.k8master
      nic_list.0.ip_endpoint_list.#: "2" => "1"


Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

nutanix_virtual_machine.k8master: Modifying... (ID: c28a4244-43b9-4aeb-88fa-800621f79939)
  nic_list.0.ip_endpoint_list.#: "2" => "1"
nutanix_virtual_machine.k8master: Still modifying... (ID: c28a4244-43b9-4aeb-88fa-800621f79939, 10s elapsed)

Error: Error applying plan:

1 error(s) occurred:

* nutanix_virtual_machine.k8master: 1 error(s) occurred:

* nutanix_virtual_machine.k8master: error waiting for vm (c28a4244-43b9-4aeb-88fa-800621f79939) to update: INTERNAL_ERROR

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

**Versions (please complete the following information):**
 - OS that is executing Terraform: macOS High Sierra
 - Terraform: Terraform v0.11.8
 - Nutanix Cluster (Prism Element / AOS) 
     AOS Version 5.8.1 
     AHV 20170830.156
     NCC Version: 3.6.0
     LCM Version: 1.4.1810
 - Nutanix Prism Central
 - Terraform provider version (using posted RC1? Compiling yourself? If so, from Develop or Master, and what commit id?
commit d2bde8ac56a29d0417c86879023c2f30302c290b (HEAD -> develop, origin/develop, origin/HEAD)

**Additional context**
```bash
resource "nutanix_virtual_machine" "k8master" {
  count = "${var.master_count}"

  name                 = "bart-k8master0${count.index + 1}"
  description          = "${var.vm_description}"
  num_vcpus_per_socket = 1
  num_sockets          = 1
  memory_size_mib      = 2048

  guest_customization_cloud_init_user_data = "${base64encode(file("cloud-init.conf"))}"

  cluster_reference = {
    kind = "cluster"
    uuid = "${var.cluster_id}"
  }

  nic_list = [{
    subnet_reference = {
      kind = "subnet"
      uuid = "${data.nutanix_subnet.external.metadata.uuid}"
    }

    ip_endpoint_list = {
      ip   = "${var.ip_start}${88+count.index}"
      type = "ASSIGNED"
    }
  }]

  disk_list = [{
    data_source_reference = [{
      kind = "image"
      uuid = "${data.nutanix_image.centos.id}"
    }]

    device_properties = [{
      device_type = "DISK"
    }]

    disk_size_mib = 5000
  }]
}
```

### PR DESCRIPTION
**Pull request by marinsalinas**
closes #12  #17 #19 

We added the nic_list_status attribute (computed) to show all the nic_list with the ip_endpoint_list from the Status object.

You could get the IP address by this:

```hcl
output "ip_address" {
  value = "${lookup(nutanix_virtual_machine.vm.nic_list.0.ip_endpoint_list[0], "ip")}"
}
```

### CODE REVIEW

File: nutanix/resource_nutanix_virtual_machine.go [hunk 1]
```diff
@@ -181,6 +181,86 @@ func resourceNutanixVirtualMachine() *schema.Resource {
...
+							},
+						},
+						"subnet_uuid": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"subnet_name": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
```
Comment (thetonymaster) [inline nutanix/resource_nutanix_virtual_machine.go:255]: All of this attributes are only computed right?

File: nutanix/resource_nutanix_virtual_machine.go [hunk 2]
```diff
@@ -737,7 +813,10 @@ func resourceNutanixVirtualMachineRead(d *schema.ResourceData, meta interface{})
 	if err := d.Set("availability_zone_reference", flattenReferenceValues(resp.Status.AvailabilityZoneReference)); err != nil {
 		return fmt.Errorf("error setting availability_zone_reference for Virtual Machine %s: %s", d.Id(), err)
 	}
-	if err := d.Set("nic_list", flattenNicList(resp.Status.Resources.NicList)); err != nil {
+	if err := d.Set("nic_list", flattenNicList(resp.Spec.Resources.NicList)); err != nil {
```
Comment (thetonymaster) [inline nutanix/resource_nutanix_virtual_machine.go:826]: If `nic_list_status` contains the output of the request and `nic_list` contains the arguments to create the nics, I think you should not set them here `nic_list`

**Comment by thetonymaster** [comment]
Also, can you add a test case for this?

**Comment by marinsalinas** [comment]
> Also, can you add a test case for this?

Yes I'd already added a test case for subnets/NICs in 1bf461b6ff9a49ca56798a6e052ee0b415ea896c, let me cherry pick and modify for the new configuration.

**Comment by thetonymaster** [comment]
Not gonna sign this off until we can try this on the lab environment.

**Pull request merged**